packages/engine-frontend/components/AGConnectionPortal.tsx

```diff
@@ -8,23 +8,66 @@ import {cn} from '@openint/ui/utils'
 import {R} from '@openint/util'
 import {WithConnectConfig} from '../hocs/WithConnectConfig'
 import {_trpcReact} from '../providers/TRPCProvider'
-import {ConnectButton} from './ConnectButton'
 import {ResourceDropdownMenu} from './ResourceDropdownMenu'
+import React from 'react'
+import {VERTICAL_BY_KEY} from '@openint/cdk'
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from '@openint/ui'
+import type { ConnectorConfigFilters } from '../hocs/WithConnectConfig'
+import {IntegrationSearch} from './IntegrationSearch'
+
 
 type ConnectEventType = 'open' | 'close' | 'error'
 
 export interface AGConnectionPortalProps extends UIPropsNoChildren {
   onEvent?: (event: {type: ConnectEventType; ccfgId: Id['ccfg']}) => void
 }
 
-// TODO: Wrap this in memo so it does not re-render as much as possible.
-// Also it would be nice if there was an easy way to automatically prefetch on the server side
-// based on calls to useQuery so it doesn't need to be separately handled again on the client...
-export function AGConnectionPortal({
+// Custom comparison function for React.memo
+const areEqual = (prevProps: AGConnectionPortalProps, nextProps: AGConnectionPortalProps) => {
+  console.log('areEqual', prevProps, nextProps);
+  return prevProps.onEvent === nextProps.onEvent && prevProps.className === nextProps.className;
+};
+
+// Define the component as a functional component
+const AGConnectionPortalComponent: React.FC<AGConnectionPortalProps> = ({
   onEvent,
   className,
-}: AGConnectionPortalProps) {
+}) => {
   const listConnectionsRes = _trpcReact.listConnections.useQuery({})
+
+  const [openDialog, setOpenDialog] = React.useState(true)
+
+  // This can be called by the same window like 
+  // postMessage({ type: 'triggerConnectDialog', value: false }, '*');
+  // or by the parent window like  
+  // const iframe = document.getElementById('openint-connect-iframeId');
+  // iframe?.contentWindow.postMessage({type: 'triggerConnectDialog', value: true },'*');
+
+  const handleMessage = React.useCallback((event: MessageEvent) => {
+    if (event.data.type === 'triggerConnectDialog') {
```

Comment:
I think we should start defining a `zod` schema for the message events that can be passed back and forth between iframes.

Reply:
It's ok for now as this is specific to AG

Reply:
I don't think we need something ag specific tbh, but not blocking